### PR TITLE
feat: tile rendering pre-process cooldown validation

### PR DESCRIPTION
**Pull request by melancholiai**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✔                                                                        |
| Tests added     | ✔                                                                        |
| Chore            | ✖                                                                       |

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,8 +45,8 @@
     "@aws-sdk/client-s3": "^3.53.1",
     "@godaddy/terminus": "4.9.0",
     "@map-colonies/cleanup-registry": "^1.1.0",
-    "@map-colonies/detiler-client": "^1.0.0",
-    "@map-colonies/detiler-common": "^1.0.0",
+    "@map-colonies/detiler-client": "^1.3.0-rc2",
+    "@map-colonies/detiler-common": "^1.3.0-rc2",
```
Comment (NivGreenstein) [inline package.json:49]: Need to change to the actual release?

File: src/retiler/tileProcessor.ts [hunk 2]
```diff
@@ -104,16 +122,21 @@ export class TileProcessor {
...
+  private async preProcess(tile: TileWithMetadata, timestamp: number): Promise<PreProcessReult> {
+    let preProcessTimerEnd;
+    let result: PreProcessReult = { shouldSkipProcessing: false };
 
-    if (this.detiler === undefined || isForced) {
-      return false;
-    }
+    try {
+      // check for forced rendering or if detiler option is off
+      const isForced = this.forceProcess || tile.force === true;
```
Comment (NivGreenstein) [inline src/retiler/tileProcessor.ts:131]:
```suggestion
      const isForced = this.forceProcess || tile.force;
```

File: src/retiler/tileProcessor.ts [hunk 3]
```diff
@@ -104,16 +122,21 @@ export class TileProcessor {
...
 
-    const detilerGetTimerEnd = this.tilesDurationHistogram?.startTimer({ kind: 'detilerGet' });
+      if (isForced || this.detiler === undefined) {
```
Comment (NivGreenstein) [inline src/retiler/tileProcessor.ts:133]:
```suggestion
      if (isForced || !this.detiler) {
```

Comment (NivGreenstein) [reply src/retiler/tileProcessor.ts:133]: Just code style :) You may not accept the suggetion

File: package.json (hunk 1 above)
Comment (melancholiai) [reply package.json:49]: after we check in in-net

File: src/retiler/tileProcessor.ts (hunk 2 above)
Comment (melancholiai) [reply src/retiler/tileProcessor.ts:131]: my linter suggest the following, "Unexpected nullable boolean value in conditional. Please handle the nullish case explicitly.eslint[@typescript-eslint/strict-boolean-expressions](https://typescript-eslint.io/rules/strict-boolean-expressions)"

File: src/retiler/tileProcessor.ts (hunk 3 above)
Comment (melancholiai) [reply src/retiler/tileProcessor.ts:133]: because its not a boolean value I prefer checking for undefined

**Review by NivGreenstein — APPROVED** [review]

**Pull request merged**